**Submit Python code**

**Perform a hypothesis test in Python**

We will use the dataset provided in the `brain_size.csv` file (Dietz et al, 2017) along with the following `scipy.stats` functions.

* `ttest_1samp`


* `vttest_ind`


* `ttest_rel`

Perform the following hypothesis tests:

1. VIQ = 0 (one-sample t-test)<br><br>

2. female_viq = male_viq (two-sample t-test)<br><br>

3. FSIQ = PIQ (paired t-test)

From Dietz et al (2017):

- Full Scale Intelligence Quotient (FSIQ)


- Verbal Intelligence Quotient (VIQ)


- Performance Intelligence Quotient (PIQ)

### Importing various libraries

In [1]:
import pandas as pd
import scipy.stats as stats

## Loading data

In [2]:
file = 'brain_size.csv'
data = pd.read_csv(file, sep=';', quotechar='"')

In [3]:
data.head()

Unnamed: 0  Gender  FSIQ  VIQ  PIQ Weight Height  MRI_Count
0           1  Female   133  132  124    118   64.5     816932
1           2    Male   140  150  124      .   72.5    1001121
2           3    Male   139  123  150    143   73.3    1038437
3           4    Male   133  129  128    172   68.8     965353
4           5  Female   137  132  134    147   65.0     951545

In [4]:
data.describe()

Unnamed: 0        FSIQ         VIQ        PIQ     MRI_Count
count   40.000000   40.000000   40.000000   40.00000  4.000000e+01
mean    20.500000  113.450000  112.350000  111.02500  9.087550e+05
std     11.690452   24.082071   23.616107   22.47105  7.228205e+04
min      1.000000   77.000000   71.000000   72.00000  7.906190e+05
25%     10.750000   89.750000   90.000000   88.25000  8.559185e+05
50%     20.500000  116.500000  113.000000  115.00000  9.053990e+05
75%     30.250000  135.500000  129.750000  128.00000  9.500780e+05
max     40.000000  144.000000  150.000000  150.00000  1.079549e+06

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Unnamed: 0  40 non-null     int64 
 1   Gender      40 non-null     object
 2   FSIQ        40 non-null     int64 
 3   VIQ         40 non-null     int64 
 4   PIQ         40 non-null     int64 
 5   Weight      40 non-null     object
 6   Height      40 non-null     object
 7   MRI_Count   40 non-null     int64 
dtypes: int64(5), object(3)
memory usage: 2.6+ KB


In [6]:
# Checking for missing values in the dataset
missing_values = data.isnull().sum()
missing_values

Unnamed: 0    0
Gender        0
FSIQ          0
VIQ           0
PIQ           0
Weight        0
Height        0
MRI_Count     0
dtype: int64

# Hypothesis Test 1: VIQ = 0 (One-sample t-test)


### Purpose: Test if the average Verbal Intelligence Quotient (VIQ) is significantly different from 0


In [18]:
viq_test_result = stats.ttest_1samp(data['VIQ'], 0)

In [19]:
viq_test_result

TtestResult(statistic=30.08809997084933, pvalue=1.3289196468727975e-28, df=39)

In [20]:
# result
test_statistic, p_value = viq_test_result
print("One-sample t-test for VIQ = 0:")
print(f"Test Statistic: {test_statistic:.2f}")
print(f"P-value: {p_value:.2e}")

One-sample t-test for VIQ = 0:
Test Statistic: 30.09
P-value: 1.33e-28


In [21]:
# Interpretation based on the p-value
alpha = 0.05  # Set your significance level

if p_value < alpha:
    print("Result: The average VIQ is significantly different from 0 at a 5% significance level.")
    print("Conclusion: There is enough evidence to reject the null hypothesis.")
else:
    print("Result: There is no significant difference between the average VIQ and 0 at a 5% significance level.")
    print("Conclusion: Fail to reject the null hypothesis. Insufficient evidence to claim a significant difference.")


Result: The average VIQ is significantly different from 0 at a 5% significance level.
Conclusion: There is enough evidence to reject the null hypothesis.


### Test Statistic: 30.09
### P-value: 1.33e-28
    
### A p-value of 1.33e-28 (which is 0.0000000000000000000000000133) is extremely small, almost negligible.
### common threshold to reject the null hypothesis is 0.05. Since 1.33e-28 is far below this threshold, it indicates we would reject the null hypothesis and conclude that the average VIQ is significantly different from 0.

### RESULT OF HYPOTHESIS TEST 1:
### REJECT NULL HYPOTHESIS, The average VIQ is significantly different from 0.


# Hypothesis Test 2: Female VIQ = Male VIQ (Two-sample t-test)


### Purpose: Test if there is a significant difference in average VIQ scores between females and males


In [22]:
# Performing the two-sample t-test for Female VIQ = Male VIQ and printing the results
female_viq = data[data['Gender'] == 'Female']['VIQ']
male_viq = data[data['Gender'] == 'Male']['VIQ']
viq_gender_test_result = stats.ttest_ind(female_viq, male_viq, nan_policy='omit')

In [23]:
#result
test_statistic, p_value = viq_gender_test_result
print("Two-sample t-test for Female VIQ = Male VIQ:")
print(f"Test Statistic: {test_statistic:.2f}")
print(f"P-value: {p_value:.2e}")

Two-sample t-test for Female VIQ = Male VIQ:
Test Statistic: -0.77
P-value: 4.45e-01


In [24]:
# Interpretation based on the p-value
alpha = 0.05  # Set your significance level

if p_value < alpha:
    print("Result: The p-value is less than 0.05.")
    print("Conclusion: There is a significant difference in average VIQ scores between females and males.")
else:
    print("Result: The p-value is greater than or equal to 0.05.")
    print("Conclusion: There is no significant difference in average VIQ scores between females and males.")


Result: The p-value is greater than or equal to 0.05.
Conclusion: There is no significant difference in average VIQ scores between females and males.


### Test Statistic: -0.77
### P-value: 4.45e-01

### Interpretation: The p-value is significantly above the conventional threshold of 0.05. This indicates that there is no significant difference in average VIQ scores between females and males. This result suggests that within this sample, gender does not have a significant impact on VIQ scores.

### Therefore, we fail to reject null hypothesis

### RESULT OF HYPOTHESIS TEST 2:
### FAIL TO REJECT NULL HYPOTHESIS,  there is no significant difference in average VIQ scores between females and males.

# Hypothesis Test 3: FSIQ = PIQ (Paired t-test)

### Purpose: Test if there is a significant difference between Full Scale IQ (FSIQ) and Performance IQ (PIQ) in the same individuals


In [25]:
# Performing the paired t-test for FSIQ = PIQ and printing the results
fsiq_piq_test_result = stats.ttest_rel(data['FSIQ'], data['PIQ'])

In [26]:
# Display results
test_statistic, p_value = fsiq_piq_test_result
print("\nPaired t-test for FSIQ = PIQ:")
print(f"Test Statistic: {test_statistic:.2f}")
print(f"P-value: {p_value:.2e}")


Paired t-test for FSIQ = PIQ:
Test Statistic: 1.78
P-value: 8.22e-02


In [27]:
# Interpretation based on the p-value
if p_value < 0.05:
    # If the p-value is less than 0.05, we reject the null hypothesis
    print("Result: There is a significant difference between Full Scale IQ (FSIQ) and Performance IQ (PIQ) in the same individuals.")
else:
    # If the p-value is greater than or equal to 0.05, we fail to reject the null hypothesis
    print("Result: There is no significant difference between Full Scale IQ (FSIQ) and Performance IQ (PIQ) in the same individuals.")


Result: There is no significant difference between Full Scale IQ (FSIQ) and Performance IQ (PIQ) in the same individuals.


### Test Statistic: 1.78

### P-value: 8.22e-02

### Interpretation: The p-value is higher than 0.05, suggesting that there is no significant difference between Full Scale IQ (FSIQ) and Performance IQ (PIQ) in the same individuals. This finding indicates a close relationship between FSIQ and PIQ measures within this dataset, implying that these two aspects of intelligence are closely linked for the individuals in this sample. 

## RESULT OF HYPOTHESIS TEST 3:
## Fail to Reject Null Hypothesis:  there is no significant difference between Full Scale IQ (FSIQ) and Performance IQ (PIQ) in the same individuals.

### OVERALL INTERPRETATION and CONCLUSION:

## One-sample t-test (VIQ not equal to 0):

The one-sample t-test on VIQ scores was conducted to determine if the average VIQ score in our group significantly differs from 0. The very small p-value (below 0.05) suggests that the average VIQ score in our group is definitely not 0. This is reasonable, as a VIQ score of 0 is highly unlikely and would indicate a complete absence of verbal intelligence.

## Two-sample t-test (Comparing VIQ in Females and Males):

A two-sample t-test was performed to assess whether there is a significant difference in average VIQ scores between females and males in the group. The p-value being more than 0.05 implies that there is no significant difference in average VIQ scores between females and males. This indicates that, in this group, being female or male doesn't substantially impact VIQ scores.

## Paired t-test (Comparing FSIQ and PIQ):

For this analysis, a paired t-test was conducted on Full Scale IQ (FSIQ) and Performance IQ (PIQ) scores within the same individuals. The p-value higher than 0.05 suggests that there's no substantial difference between FSIQ and PIQ scores. This may imply that FSIQ and PIQ scores in this group measure similar aspects of intelligence. 